### PR TITLE
Improved Documentation

### PR DESCRIPTION
**Pull request by FelixTheilen**
This pull request has been redesigned to specifically adress issue https://github.com/desy-ml/cheetah/issues/2 .
-The pull request introduces a new comment on the 'init' function. This will help finding the definitions for the parameters.
-Also a new comment on the 'transfer_map' function has been added, explaining the general purpose as well as the state vector and parameters, returns and raises.
-The parameter 'energy' has been removed from various classes. The parameter 'energy' is passed on through the beam passed to the segment. It is therefore not changeable by the user, when trying to control the beam, e.g. you can't call the function HorizontalCorrector and pass a parameter energy to this function. This parameter is defined, when the beam is fed in the Segment.
-'repr' has now a short comment describing the use of this function.
-Spelling mistakes have been corrected in accelerator.py and particles.py.

### CODE REVIEW

**Comment by cr-xu** [comment]
Hi, I marked the PR as draft. Once all changes are made and pushed, you can mark as ready and as for review/merge.

**Comment by jank324** [comment]
@FelixTheilen I think it makes sense to split this pull request into three:
 - Extension of and spelling fixes for the documentation (addressing issue #2)
 - Fix for BPM bug (addressing issue #20)
 - Fix for uninformative warning about `ParameterBeam` not being on the same device as a `Segment` when a GPU is available (addressing issue #19)

File: cheetah/accelerator.py [hunk 1]
```diff
@@ -21,6 +21,10 @@
 class Element:
     """
     Base class for elements of particle accelerators.
+    Every subclass depending on this base class inherits the Parameters, Attributes and Functions from this base class.
+    New Parameters, Attributes and Functions can be added in the subclass and the Functions can be redefined. 
+    To sum up the base class is used to set the default settings for every subclass, as well as commenting on the behavior of 
+    various base functions, which are just slightly changed for each subclass.
```
Comment (jank324) [inline cheetah/accelerator.py:27]: I think this comment is not needed as that is how classes work in Python and other programming languages, i.e. not specific to Cheetah.

File: cheetah/accelerator.py [hunk 2]
```diff
@@ -62,6 +62,24 @@ def __init__(self, name=None, device="auto"):
         self.device = device
 
     def transfer_map(self, energy):
+        """
+        Necessary for the calculations in '__call__'.
```
Comment (jank324) [inline cheetah/accelerator.py]: Maybe be be more specific along the lines of "Generates the element's transfer map that describes how the beam and its particles are transformed when travelling through the element.

File: cheetah/accelerator.py [hunk 3]
```diff
@@ -63,7 +68,16 @@ def __init__(self, name=None, device="auto"):
...
+        Necessary for the calculations in '__call__'. The Matrix R that is returned is then multiplied with the state vector.
+        The state vector is consisting of 6 values with a physical meaning:
+            x: Position in x direction
+            xp: Momentum in x direction
+            y: Position in y direction
+            yp: Momentum in y direction
+            s: Position in z direction, the zero value is set to the middle of the pulse
+            sp: Momentum in s direction
+        As well as a seventh value used to add constants to some of the prior values if necesassary. Through this
+        seventh state, the addition of constants can be represented using a matrix multiplication.
```
Comment (jank324) [inline cheetah/accelerator.py:80]: I like that you added this 👌🏻

File: cheetah/particles.py [hunk 4]
```diff
@@ -238,7 +238,7 @@ def __init__(self, mu, cov, energy):
         self._mu = mu
         self._cov = cov
         self.energy = energy
-        if element.device == "auto" and torch.cuda.is_available():
+        if element.device == "cuda":
```
Comment (jank324) [inline cheetah/particles.py:241]: We talked about this. I think it's better to check in `__call__` if the beam's device matches the segment's device and issue a meaningful exception when it doesn't.

**Comment by FelixTheilen** [comment]
This pull request has been redesigned to specifically adress issue #2 . 
-The pull request introduces a new comment on the '__init__' function. This will help finding the definitions for the parameters. 
-Also a new comment on the 'transfer_map' function has been added, explaining the general purpose as well as the state vector and parameters, returns and raises.
-The parameter 'energy' has been removed from various classes. The parameter 'energy' is passed on through the beam passed to the segment. It is therefore not changeable by the user, when trying to control the beam, e.g. you can't call the function HorizontalCorrector and pass a parameter energy to this function. This parameter is defined, when the beam is fed in the Segment.
-'__repr__' has now a short comment describing the use of this function.
-Spelling mistakes have been corrected in accelerator.py and particles.py.

**Comment by FelixTheilen** [comment]
I reused the branch needed for the improvement of the documentation. This resulted in this pull request not changing anything in the desy-ml/master file. Unfortunately this couldnt be reverted, as i had the previous file, but i could'nt update since Github thought the removal of items made the master branch more up to date.

**Pull request closed without merging**